aws-gitlab-main.yml

```diff
@@ -0,0 +1,214 @@
+        Properties:
+          SubnetId: !Ref gitlab-public-10.0.2.0
+          RouteTableId: !Ref gitlab-public-RouteTable
+
+      gitlab-private-RouteTable-a:
+        Type: 'AWS::EC2::RouteTable' 
+        Properties:
+          VpcId: !Ref gitlabVPC
+      
+      gitlab-private-Route-a:
+        Type: 'AWS::EC2::Route' 
+        Properties:
+          RouteTableId: !Ref gitlab-private-RouteTable-a
+          DestinationCidrBlock: 0.0.0.0/0
+      
+      gitlab-private-10.0.1.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-private-10.0.1.0
+          RouteTableId: !Ref gitlab-private-RouteTable-a 
+
+      gitlab-private-RouteTable-b:
+            Type: 'AWS::EC2::RouteTable' 
+            Properties:
+              VpcId: !Ref gitlabVPC
+          
+      gitlab-private-Route-b:
+        Type: 'AWS::EC2::Route' 
+        Properties:
+          RouteTableId: !Ref gitlab-private-RouteTable-b
+          DestinationCidrBlock: 0.0.0.0/0
+
+      gitlab-private-10.0.3.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-private-10.0.3.0
+          RouteTableId: !Ref gitlab-private-RouteTable-b
+
+      #LOAD BALANCER
+      gitlab-LoadBalancer:
+        Type: 'AWS::ElasticLoadBalancing::LoadBalancer'
+        Properties: 
+          Listeners: 
+          - LoadBalancerPort: '80'
+            InstancePort: '80'
+            Protocol: HTTP
+          - LoadBalancerPort: '443'
+            InstancePort: '443'
+            Protocol: HTTPS
+          - LoadBalancerPort: '22'
+            InstancePort: '22'
+            Protocol: TCP
+          HealthCheck: #not sure about this section
+            Target: HTTP:80/
+            #HealthyThreshold: '3'
+            #UnhealthyThreshold: '5'
+            #Interval: '30'
+            #Timeout: '5'
+          #Req:"Ping Protocol (HTTP), Ping Port (80), Ping Path (/users/sign_in)"
+          LoadbalancerName: gitlab-LoadBalancer
+          Policies:
+            #??? 
+          Scheme: internet-facing #?
+          Security Groups: #not sure about this section
+            gitlab-SecurityGroup:
+              Type: AWS::EC2::SecurityGroup
+              Properties:
+                GroupName: gitlab-loadbalancer-sec-group 
+                GroupDescription: Allow HTTP and HTTPS traffic from anywhere 
+                VpcId: !Ref gitlabVPC
+                SecurityGroupIngress: 
+                - IpProtocol: HTTP
+                FromPort: 80
+                ToPort: 80
+                CidrIp: 0.0.0.0/0 
+                -IpProtocol: HTTPS
+                FromPort: 443
+                ToPort: 443
+                CidrIp: 0.0.0.0/0
+          #Other reqs I'm not sure how to address:
+          #allow SSH traffic, selct a custom source, add single trusted IP address or IP address range in CIDR notation
```

Comment:
Add port 22 as part of security group. The place where it says cidr you would include your own public ip or trusted ip set otherwise you open this to world

Reply:
Make port 22 security group and parametrize the cidr